noir-projects/noir-protocol-circuits/crates/types/src/meta/mod.nr

```diff
@@ -98,7 +98,14 @@ pub comptime fn flatten_to_fields(name: Quoted, typ: Type, omit: [Quoted]) -> ([
             let nested_struct = typ.as_struct().unwrap();
             let params = nested_struct.0.fields();
             let struct_flattened = params.map(
-                | (param_name, param_type): (Quoted, Type) | flatten_to_fields(quote {$name.$param_name}, param_type, omit)
+                | (param_name, param_type): (Quoted, Type) | {
```

Comment:
This change was introduced to handle the case where the resulting quotes are not prefixed with `self.`. This was needed because I have the values to be serialized passed as args to a function instead of having them be a members of a struct.